lib/core/src/Cardano/Wallet/Api/Server.hs

```diff
@@ -2214,9 +2217,15 @@ balanceTransaction
     -> Handler (ApiConstructTransaction n)
 balanceTransaction ctx genChange (ApiT wid) body = do
     pp <- liftIO $ NW.currentProtocolParameters nl
+    pp' <- (error "TODO: balanceTransaction in byron")
+        <$> liftIO (NW.currentNodeProtocolParameters nl)
+
+    -- TODO: Check that sealedTxIncoming contains a PParam hash matching both pp
+    -- and pp'. Without this, collateral could be forfeited!
```

Comment:
I am not sure to understand this one? Note that I took care of adding the script integrity hash when necessary as part of #2952

Reply:
Ah, thanks, indeed. If we always set the script integrity hash, then it should be nothing to worry about. I thought it looked like in [cardano-api](https://github.com/input-output-hk/cardano-node/blob/0ed04ab8d0a205691219f04623a3d880545a3f34/cardano-api/src/Cardano/Api/TxBody.hs#L1192) (but I now see it's only used to derive the script integrity hash), and that you said PAB would set it (https://github.com/input-output-hk/cardano-wallet/pull/2952#discussion_r723914685)